### PR TITLE
Adds a proof harness for the IotMqtt_strerror function

### PR DESCRIPTION
**Pull request by feliperodri**
*Description of changes:*

This PR depends on https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/788!

- Adds a new proof harness for the `IotMqtt_strerror` function.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cbmc/proofs/IotMqtt_strerror/IotMqtt_strerror_harness.c [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#include "iot_config.h"
+#include "private/iot_mqtt_internal.h"
+
+void harness()
+{
+  IotMqttError_t status;
+  const char *pMessage = IotMqtt_strerror(status);
+  assert(pMessage != NULL);
```
Comment (markrtuttle) [inline cbmc/proofs/IotMqtt_strerror/IotMqtt_strerror_harness.c]: Nice.

File: cbmc/proofs/IotMqtt_strerror/IotMqtt_strerror_harness.c [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+
+
+
```
Comment (danielsn) [inline cbmc/proofs/IotMqtt_strerror/IotMqtt_strerror_harness.c]: extra whitespace

File: cbmc/proofs/IotMqtt_strerror/cbmc-batch.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+jobos: ubuntu16 
+cbmcflags:        '=--unwind;1;--bounds-check;--pointer-check;--unwinding-assertions;--nondet-static=' 
+goto: IotMqtt_strerror.goto
+expected: SUCCESSFUL
```
Comment (danielsn) [inline cbmc/proofs/IotMqtt_strerror/cbmc-batch.yaml]: sort. also whitespace

File: cbmc/proofs/IotMqtt_strerror/cbmc-batch.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+jobos: ubuntu16 
+cbmcflags:        '=--unwind;1;--bounds-check;--pointer-check;--unwinding-assertions;--nondet-static='
```
Comment (danielsn) [inline cbmc/proofs/IotMqtt_strerror/cbmc-batch.yaml]: Seems like you're missing flags?

File: cbmc/proofs/IotMqtt_strerror/IotMqtt_strerror_harness.c (hunk 2 above)
Comment (feliperodri) [reply cbmc/proofs/IotMqtt_strerror/IotMqtt_strerror_harness.c]: Done!

File: cbmc/proofs/IotMqtt_strerror/cbmc-batch.yaml (hunk 3 above)
Comment (feliperodri) [reply cbmc/proofs/IotMqtt_strerror/cbmc-batch.yaml]: Done!

File: cbmc/proofs/IotMqtt_strerror/cbmc-batch.yaml (hunk 4 above)
Comment (feliperodri) [reply cbmc/proofs/IotMqtt_strerror/cbmc-batch.yaml]: I forgot to update the `cbmc-batch.yaml` file.

**Pull request merged**